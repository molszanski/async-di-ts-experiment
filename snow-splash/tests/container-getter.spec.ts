import { getMainMockAppContainer } from "./mocks/_mock-app-container"

it("should get a single container", () => {
  // This is silly
  ;(async () => {
    const cont = getMainMockAppContainer()
    expect(cont.containers).toHaveProperty("bCont")
    expect(cont.containers.aCont).toBeInstanceOf(Promise)

    let b = await cont.containers.bCont
    expect(b).toHaveProperty("b2")
    expect(b).toMatchSnapshot()
  })()
})

it("should subscribe to a single container", (cb) => {
  // This is silly
  ;(async () => {
    const cont = getMainMockAppContainer()
    expect(cont.containers).toHaveProperty("bCont")
    expect(cont.containers.aCont).toBeInstanceOf(Promise)

    let cCont = await cont.containers.cCont
    cont.subscribeToContiner("cCont", (new_cCont) => {
      expect(new_cCont.c2.size).toBe(10)
      cb()
    })
    cCont.upgradeCContainer()
  })()
})
