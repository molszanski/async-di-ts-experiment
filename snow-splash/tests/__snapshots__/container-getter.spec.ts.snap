// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get a single container 1`] = `
Object {
  "b1": B1 {
    "a2": A2 {
      "a1": A1 {},
    },
  },
  "b2": B2 {
    "a1": A1 {},
  },
}
`;
